gate-keeper/README.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: pattern
+folder: gate-keeper
+permalink: /patterns/gate-keeper/
+categories: Cloud
+tags:
+ - Cloud distributed
+---
+
+## Intent
+
+Protect applications and services by using a dedicated host instance that acts as a broker between clients and the application or service, validates and sanitizes requests, and passes requests and data between them. 
+
+## Explanation
+
+Real world example
+
+> If our server will access some sensitive databases or other servers, we can create a Gatekeeper role for this server, and we use this Gatekeeper role to authenticate the requests from the client, so that the server's access to the database is hidden from the client. This can provide an additional layer of security, and limit the attack surface of the system.
+
+**Programmatic Example**
+
+We have an abstract class `Request` and two subclasses `HackRequest` and `LoginRequest`.
+
+```java
+public abstract class Request {
+  String action;
+}
+
+public class HackRequest extends Request {
+  public HackRequest(String action) {
+    super(action);
+  }
+}
+
+public class LoginRequst extends Request {
+  private String account;
+  private String password;
+
+  public LoginRequst(String action, String account, String password) {
+    super(action);
+    this.account = account;
+    this.password = password;
+  }
+}
+```
+
+And we have a `TrustRole` as server to commuicate with database and handle the requests that meet the requirements.
+
+```java
+public class TrustRole {
+  public void handleRequest(Request request) {
+    if (request.getAction().equals("login")) {
+      LOGGER.info("Handle login request.");
+      handleLogin((LoginRequst) request);
+    }
+  }
+
+  public void handleLogin(LoginRequst request) {
+    LOGGER.info("Link to the database and check account: "
+            + request.getAccount() + " and password: " + request.getPassword() + ".");
+  }
+
+}
+```
+Then we need to add a `GateKeeper` to verify these request, and choose the requests that meet the requirements to forward to the server.
+
+```java
+public class GateKeeper {
```

Comment:
What is keeping someone from creating a class that inherits from GateKeeper and overrides the checkRequest method with return true? They can use that new class to work around any validation checks

Reply:
The gatekeeper class is on the server side and is used to process incoming commands from the client, the client itself does not have access to this class

Reply:
The code shows both classes being called in the same code. The client can just extend the Gatekeeper class and make validations pass.